src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -135,7 +135,30 @@ async Task OnBind(ListenOptions options)
                     }
 
                     var connectionDispatcher = new ConnectionDispatcher(ServiceContext, connectionDelegate);
-                    var transport = await _transportFactory.BindAsync(options.EndPoint).ConfigureAwait(false);
+
+                    IConnectionListenerFactory factory = null;
+                    if (options.Protocols == HttpProtocols.Http3)
+                    {
+                        foreach (var transportFactory in _transportFactories)
+                        {
+                            if (transportFactory.GetType().Name.Contains("Quic"))
```

Comment:
Can we add the marker interface this PR as well?

Reply:
Which interface? IMultiplexedConnectionTransportFactory? I'm writing an API proposal for that now.